### PR TITLE
Misc Consul health checks

### PR DESCRIPTION
**Pull request by bwester**
Some bug fixes and improvements I've been working on related to Consul's health checking

* Fix faulty Consul health check
* Create Consul health checks with explicit hostname
* Add feature flag for Consul health checks

### CODE REVIEW

**Comment by anthonybishopric** [comment]
:+1:

File: pkg/kp/service.go [hunk 1]
```diff
@@ -26,14 +27,27 @@ func (c consulStore) RegisterService(manifest pods.Manifest, caPath string) erro
...
+			return err
+		}
 		podService.Port = manifest.StatusPort
 		podService.Check = &api.AgentServiceCheck{
 			Interval: checkInterval,
 		}
 		if manifest.StatusHTTP {
-			podService.Check.Script = fmt.Sprintf(httpStatusCheck, manifest.StatusPort)
+			podService.Check.Script = fmt.Sprintf(
+				hostname,
```
Comment (petertseng) [inline pkg/kp/service.go]: I am very surprised that `hostname` is coming before `httpStatusCheck`! is there some magic I am no aware of to make this work?

File: pkg/kp/service.go [hunk 2]
```diff
@@ -26,14 +27,27 @@ func (c consulStore) RegisterService(manifest pods.Manifest, caPath string) erro
...
+				httpStatusCheck,
+				manifest.StatusPort,
+			)
 		} else {
-			podService.Check.Script = fmt.Sprintf(httpsStatusCheck, manifest.StatusPort, caPath)
+			podService.Check.Script = fmt.Sprintf(
+				hostname,
```
Comment (petertseng) [inline pkg/kp/service.go]: same question w/ `hostname` before `httpsStatusCheck`

File: pkg/kp/service.go (hunk 1 above)
Comment (bwester) [inline pkg/kp/service.go]: Nope, this definitely a bug, the fix of which did not make it into the PR.

Comment (bwester) [inline pkg/kp/service.go]: Fix is in: https://github.com/square/p2/pull/212

**Pull request merged**